### PR TITLE
Improve task list

### PR DESCRIPTION
**Pull request by andysellick**
- fix issues with number vertical alignment and panel top spacing on mobile
- add click tracking for panel links
- code update to rename the parts of a task list, now contains groups each containing steps, instead of steps containing sections

Before:

screen shot 2017-11-02 at 16 26 53

After:

screen shot 2017-11-02 at 16 31 33

Component guide link: https://govuk-static-pr-1182.herokuapp.com/component-guide/task_list

Trello card: https://trello.com/c/hjuxqyuT/275-visual-and-code-tweaks-to-the-task-list

Tested on iphone 6s, Google Nexus 6, iphone 6, Samsung Galaxy s5, ipad air 2, Google nexus 7, Samsung galaxy tab 4, Google Pixel
IE8, IE9, IE10, FF 56, Opera 48, Chrome latest, Safari 5.1, Edge 16

### CODE REVIEW

**Comment by vanitabarrett** [comment]
The new vertical spacing fix on mobile looks different to what you get normally on desktop? Should these be vertically centred too?

screen shot 2017-11-03 at 09 59 06

**Comment by andysellick** [comment]
@vanitabarrett Good spot, no you're right, the numbers should always be vertically centered. What browser/device/screen size/task list was that?

**Comment by vanitabarrett** [comment]
@andysellick Just Chrome on Mac, viewing the small task list in the component guide (herokuapp link in PR above).

**Comment by andysellick** [comment]
@vanitabarrett odd that it looked fine for me. Have had another crack at it, very difficult to get it perfect cross browser (might be slightly off on iphone but little I can do about that), but should be better now.

**Comment by vanitabarrett** [comment]
This looks good to me 👍

File: app/views/govuk_component/docs/task_list.yml [hunk 1]
```diff
@@ -84,7 +84,7 @@ examples:
     description: Sections have a H2 by default, but this can be changed. The heading level does not change any styling.
```
Comment (sihugh) [inline app/views/govuk_component/docs/task_list.yml]: Should this read `Steps have a H2 by default...`?

File: app/views/govuk_component/task_list.raw.html.erb [hunk 2]
```diff
@@ -1,73 +1,75 @@
 <%
   steps ||= false
+  groups ||= steps #changing the name of this attribute, temporarily accepting both to avoid deployment problems
```
Comment (sihugh) [inline app/views/govuk_component/task_list.raw.html.erb]: 👍

File: app/views/govuk_component/docs/task_list.yml (hunk 1 above)
Comment (andysellick) [reply app/views/govuk_component/docs/task_list.yml]: Good spot, thanks.

**Pull request merged**